cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -649,7 +716,8 @@ fn insert_common_prologue(
             // Also, the size of a return address, implicitly pushed by a x86 `call` instruction,
             // also should be accounted for.
             // TODO: Check if the function body actually contains a `call` instruction.
-            let total_stack_size = (csrs.iter(GPR).len() + 1 + 1) as i64 * word_size as i64;
+            let total_stack_size = (csrs.iter(GPR).len() + 1 + 1) as i64 * word_size as i64
+                + csrs.iter(FPR).len() as i64 * types::F64X2.bytes() as i64;
```

Comment:
`csrs.iter(FPR).len()` will be zero via the above assert, but I want the stack size to be the same expression in the three places we compute it here in `abi.rs`.

Reply:
I think unfortunately that this isn't quite right -- in order to store the xmm registers, the destination must be 16-byte aligned. Since the number of GPRs pushed is non-constant, you might need padding bytes.

Reply:
I'd hoped to store XMM registers before worrying about GPRs, so they would be a constant offset rather than conditionally requiring padding. I don't think that actually pans out though, and it's still variable on architecture width, so I'll adjust this to include some padding space if `csrs.iter(FPR).len() != 0`.